### DISCUSSION TITLE
How to load addons/extensions?

### DISCUSSION THREAD

**Discussion by DaPotatoMan** [tauri-apps/tauri#2685] [Q&A] [answered]
Hello.
Is there any way to load extensions in the webview (possibly with javascript)?

**Comment by FabianLars** [comment, marked as answer]
do you mean extensions as in "browser extensions" (like the ones from the chrome webstore)? If so then the answer is no, not possible.

Edit: Just to clear things up, this isn't something tauri can "fix", the WebViews themselves don't support this.

**Reply by DaPotatoMan** [reply]
Got it. Thanks for replying.

**Reply by ctwhome** [reply]
what a pity, that would bring the whole ecosystem to these kind of multiplatform apps

**Reply by hathiphant** [reply]
Isn't it sometimes possible?

https://mozilla.github.io/geckoview/javadoc/mozilla-central/org/mozilla/geckoview/GeckoRuntime.html#getWebExtensionController()

Thanks,

**Reply by FabianLars** [reply]
We don't use gecko/firefox on any of our supported platforms so that link doesn't mean anything for tauri.

That said, webview2, tauri's webview on Windows added support for it recently([docs](https://learn.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2profile7?view=webview2-1.0.2210.55#addbrowserextension)) though the webview2-com bindings crate wasn't updated yet (once it is, it will only work in tauri v2). Also right now t i kinda doubt we'll have an explicit function for this in tauri in the near future, but i'm fairly sure it can be done in apps with the existing resources feature and the [`with_webview`](https://docs.rs/tauri/latest/tauri/window/struct.Window.html#method.with_webview) api.

**Reply by kylinholmes** [reply]
How about write a browser-side extension that loads the js code from tauri-side, and to be executed when the tauri is launched, Like a `Loader`.

**Reply by Haprog** [reply]
I'm not sure if this is about the same thing but now there is this https://github.com/tauri-apps/tauri/issues/10909 (Windows only feature). But I'm not sure if there is any documentation for this yet?

**Comment by hhstore** [comment]
- https://github.com/rubickCenter/rubick
   - https://rubickcenter.github.io/rubick/dev/
   - plugin API : https://rubickcenter.github.io/rubick/api/
- Implement similar functionality to this project:  
   - 1. Implement a plugin store that can load plugins at runtime. 
   - 2. Plugins can be implemented using JS/TS, or other languages, such as dart (dart can be compiled into js) 
   - 3. Implement some hook functions, preload, afterload, etc., and provide some APIs for plugin calls.
- To implement this requirement, may you need to use the [WebView](https://github.com/tauri-apps/wry), [window](https://tauri.app/v1/api/js/window/), [events](https://tauri.app/v1/guides/features/events/) or the [sidecar](https://tauri.app/v1/guides/building/sidecar/) pattern?
- List of related `Tauri APIs`:
    - [event](https://tauri.app/v1/api/js/event) 
    - [shell](https://tauri.app/v1/api/js/shell)
    - [window](https://tauri.app/v1/api/js/window)
    - [http](https://tauri.app/v1/api/js/http)

**Comment by 7flash** [comment]
Are you proposing support of chosen plugins at build time by developer or any user installed plugins at run time?

**Comment by reyamir** [comment]
Hi, I'm looking for some tips to implement plugin system at runtime, similar to Raycast or VSCode extension,... any ideas?

**Comment by holybaechu** [comment]
You can still use `eval` or `initalization_script` to inject scripts. In that way, You can load some extensions

**Comment by ZeeshanAhmadKhalil** [comment]
I someone is trying to use [redux-devtools](https://chromewebstore.google.com/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd) extension to debug redux then it's alternative is to use [redux-logger](https://www.npmjs.com/package/redux-logger#install) package.
